lib/active_model/serializer.rb

```diff
@@ -168,18 +147,6 @@ def json_key
       root || object.class.model_name.to_s.underscore
     end
 
-    def attributes
-      attributes = self.class._attributes.dup
-
-      attributes.each_with_object({}) do |name, hash|
-        if self.class._fragmented
-          hash[name] = self.class._fragmented.public_send(name)
-        else
-          hash[name] = send(name)
-        end
-      end
-    end
-
```

Comment:
funny, I see this as **the** fundamental concern of the serializer and probably the only thing I **wouldn't** remove (though maybe I'd extract method object)